Add 0.317.0 release notes (#238)
trinodb/trino-python-client: CHANGES.md

[README](https://github.com/trinodb/trino-python-client/blob/master/README.md)
and the [PyPI page](https://pypi.org/project/trino/).

> ## Trino Python client 0.317.0

Reviewer: How about just `Release 0.317.0`? Repeating `Trino Python client` in Trino Python client looks redundant. 

No need to change in this PR since `Trino Python client 0.316.0` already exists.

[README](https://github.com/trinodb/trino-python-client/blob/master/README.md)
and the [PyPI page](https://pypi.org/project/trino/).

## Release 0.317.0

* Add support for creating tables containing `JSON` columns and reading and
  writing to them with SQLAlchemy.
  ([#194](https://trinodb/trino-python-client/issues/194))
* Properly propagate query failures to the client when using `fetchone`.
  ([#95](https://trinodb/trino-python-client/issues/95))
* Fix queries returning a single row from sometimes appearing as failed on the
  server. ([#220](https://trinodb/trino-python-client/issues/220))
* Fix query failures when using SQLAlchemy `Table` without a catalog name

>   provided. ([#237](https://trinodb/trino-python-client/issues/237))

Author: @mdesmet Does this correctly capture the user-observed bug underlying #237 (fixed in #240)?

Author: ping @mdesmet

[README](https://github.com/trinodb/trino-python-client/blob/master/README.md)
and the [PyPI page](https://pypi.org/project/trino/).

## Release 0.317.0

* Add support for creating tables containing `JSON` columns and reading and
  writing to them with SQLAlchemy.
  ([#194](https://trinodb/trino-python-client/issues/194))

> * Properly propagate query failures to the client when using `fetchone`.

Reviewer 2: I'd generally word this as "Fix query failures not being propagated to the client when..."

I like to start with the words Add/Fix/Improve as much as possible so it's abundantly clear what each type of change is.

  ([#95](https://trinodb/trino-python-client/issues/95))
* Fix queries returning a single row from sometimes appearing as failed on the
  server. ([#220](https://trinodb/trino-python-client/issues/220))

> * Fix query failures when using SQLAlchemy `Table` without a catalog name
>   provided. ([#237](https://trinodb/trino-python-client/issues/237))

Reviewer 3: There is no way to provide a catalog to a `TableClause`. 

```suggestion
* Fix query failures when using SQLAlchemy `TableClause` by not performing
catalog lookup. ([#237](https://trinodb/trino-python-client/issues/237))
```

* Fix errors when using prepared statements with Trino versions greater than or
  equal to 398. ([#242](https://trinodb/trino-python-client/issues/242))

### Breaking Changes

> * Make the `execute` method of the cursor block until at-least one row is
>  received. This means users no longer need to call `fetchone` or `fetchall` to
>  make sure query actually starts executing on the Trino server. Note that
>  results still need to be consumed by calling `fetchone` or `fetchall` to make
>  sure query isn't considered idle and terminated on the server.
>  ([#232](https://trinodb/trino-python-client/issues/232))

Reviewer 2: ```suggestion
* Block the `execute` method of the cursor until at least one row is
 received. Users no longer need to call `fetchone` or `fetchall` to
 ensure that a query starts executing on the Trino server. Note that
 results still need to be consumed by calling `fetchone` or `fetchall` to
 ensure that a query isn't considered idle and terminated on the server.
 ([#232](https://trinodb/trino-python-client/issues/232))
```

Reviewer 2: No hyphen in at least, a few wording changes, and making sure we're properly using a/an/the.

* Add a function `trino.sqlalchemy.URL` to generate SQLAlchemy URLs which
  properly handles escaping and encoding values where needed.
  ([#235](https://trinodb/trino-python-client/issues/235))
* Fix query failures not being propagated to the client when using `fetchone`.
  ([#95](https://trinodb/trino-python-client/issues/95))
* Fix queries returning a single row from sometimes appearing as failed on the
  server. ([#220](https://trinodb/trino-python-client/issues/220))
* Fix query failures when using SQLAlchemy `TableClause` by not performing
  catalog lookup. ([#237](https://trinodb/trino-python-client/issues/237))
* Fix errors when using prepared statements with Trino versions greater than or
  equal to 398. ([#242](https://trinodb/trino-python-client/issues/242))

### Breaking Changes

* Block the `execute` method of the cursor until at least one row is received.
  Users no longer need to call `fetchone` or `fetchall` to ensure query starts
  executing on the Trino server. Note that results still need to be consumed by
  calling `fetchone` or `fetchall` to ensure that a query isn't considered idle
  and terminated on the server.

>   ([#232](https://trinodb/trino-python-client/issues/232))

Reviewer: Could you change all URL to contain `github.com`? 
`https://trinodb/trino-python-client/issues/232` → `https://github.com/trinodb/trino-python-client/issues/232`

Author: Thanks for catching this. 🤦
